hpcgap/src/gvars.c

```diff
@@ -1071,28 +1116,29 @@ static Int NCopyAndFopyGVars;
 **  This is OK for garbage collection, but  a real problem  for saving in any
 **  event, this information  does not really want to  be saved  because it is
 **  kernel centred rather than workspace centred.
-**
-**  Accordingly we     provide  two    functions    `RemoveCopyFopyInfo'  and
-**  `RestoreCopyFopyInfo' to  remove  or restore   the  information from  the
-**  workspace.  The  Restore  function is  also   intended to  be used  after
-**  loading a saved workspace
 */
 void InitCopyGVar (
     const Char *        name ,
     Obj *               copy )
 {
     /* make a record in the kernel for saving and loading                  */
+#ifdef HPCGAP
     LockGVars(1);
+#endif
     if ( NCopyAndFopyGVars >= MAX_COPY_AND_FOPY_GVARS ) {
+#ifdef HPCGAP
         UnlockGVars();
+#endif
```

Comment:
@rbehrends I was wondering whether we really need to call `UnlockGVars` before panicking out. Do we? My theory was that `Pr` might end up trying to access some gvar, and so to avoid a deadlock, we need to unlock here.
In any case, we should either remove that unlocking, or else at a comment that explains why it needs to be there (same for at least one other place with similar code).